### DISCUSSION TITLE
create zfs volume per for each podman volume?

### DISCUSSION THREAD

**Discussion by psy0rz** [containers/podman#20435] [General]
I've configured the zfs storage driver which seems to work fine for images.

However: i want a zfs volume for each podman volume? 

For docker i'm using: https://github.com/csachs/docker-zfs-plugin

Is there a way to achieve this for podman? 

Or is there a way to do this manually? (Do a zfs create for each volume or something before running podman)

Edwin

**Comment by Luap99** [comment]
Podman does support docker style volume plugins so you should be able to keep using the same plugin with podman.
The only thing is you must register the plugin in containers.conf under `[engine.volume_plugins]`, for example https://github.com/containers/podman/blob/19c870da0d4923b2bf2d62b5d424da5a649b4d40/test/e2e/config/containers.conf#L69C1-L72

**Reply by psy0rz** [reply]
oh nice ill try that.

**Reply by davidchisnall** [reply]
Please can you share whether (and how) you got this to work? I would also like this functionality.

**Reply by psy0rz** [reply]
nope, not tried it yet.

**Reply by davidchisnall** [reply]
I have not been able to make this work at all.  I have added this to my containers.conf:

```
[engine.volume_plugins]
docker-zfs-plugin = "/var/run/docker-zfs-plugin"
```

This is the path where I've put the socket for `docker-zfs-plugin`, but I can't see anything in the output of `podman --log-level trace volume create test` that even hints at it trying this.  

I also agree that this should be the default behaviour volumes on ZFS and it surprised me that it was not.  Not being able to independently snapshot, send / receive volumes feels like missing the point of ZFS.

**Reply by psy0rz** [reply]
They are at least using the snapshot and clone function to handle images, instead of overlay2. 

But indeed you would expect them to use it for volumes as well.

**Reply by Luap99** [reply]
You need `podman volume create --driver docker-zfs-plugin ...` once you configured it in containers.conf
